src/Keboola/StorageApi/Client.php

```diff
@@ -1330,10 +1331,29 @@ public function uploadFile($filePath, FileUploadOptions $options)
                 $result['name']
             );
 
+            $cnt = 0;
             do {
                 try {
+                    $cnt++;
                     $s3result = $uploader->upload();
                 } catch (MultipartUploadException $e) {
+                    $this->log('Multipart upload failed: ' . $e->getMessage());
```

Comment:
Přebírá ty logy z klienta docker runner? Pokud jo tak pro debug imho stačí první nasadit to logování, abysme věděli o co jde a v jaké míře se to děje.
Druhá varianta je jak tminoval někde najloš ten počet pokusu nastavit vysoký třeba na 1000 a doufat že se to neděje běžně a že to fakt chytí ty timeouty.

Reply:
overim to

Reply:
neprebira! a navic kdyz tam ten logger zapnem, tak to bude logovat kazdej potapanej API request https://github.com/keboola/storage-api-php-client/blob/master/src/Keboola/StorageApi/Client.php#L146

Reply:
mozna by se tady https://github.com/keboola/output-mapping/blob/master/Writer/Writer.php#L69 dal udelat clone clienta a zapnout ten logger jen pro ten OM ?

Reply:
i tak by se to ale asi melo zmenit na level notice, jinak to bude strasit v storage eventech :/

Reply:
hmm, to logování by asi trochu chtělo předělat aby bylo použitelný. Tomu middleware na ty requesty by se dalo nastavit ať loguje `DEBUG`. A tohle by mohl být `NOTICE` teda.
Jenže když dáš `NOTICE` tak tím pádem bude logovat i info a vyšší ne?

Reply:
tam jde o level te konkretni hlasky - to sem vnasime bohuzel nejaky implementacni detaily z docker runner - Ten loguje info a vyssi do storage i pt (debug na produkci neloguje nikam). A na eventy ktery jsou notice je tam vyhybka, ze se neposilaji do storage https://github.com/keboola/docker-bundle/blob/master/src/Service/LoggersService.php#L46

(log levels jsou debug < info < notice < warning https://github.com/Seldaek/monolog/blob/master/doc/01-usage.md#log-levels)

Reply:
cili - jo, kdyz dabe na requesty debug a na ten error notice a predame tomu ten logger, tak by to melo chodit

plus pridat search do PT, at o tom vime

Reply:
no ja to myslel tak ze kdyz tam das to at to loguje ten notice tak to najednou zacne posilat i pripadne info (a objevi se v projektech) jestli tam nejake jsou (krome tech requestu ktere se zmeni na debug).

Reply:
Myslím tím třeba tohle https://github.com/keboola/storage-api-php-client/blob/e75af8f470eafcd3875db4bfd0f706f00b1142fe/src/Keboola/StorageApi/Client.php#L260

Reply:
jo, no ja to "predelat logovani" myslel, tak, ze dame asi vicemene vsechno co je ted `log` jako `debug` a ten multipart jako `notice`. V kombinaci s tim, ze to zapnem jen pro ten OM by to teda nemelo produkovat nic do eventu projektu

Reply:
jo tak pak cajk

Reply:
tam jeste rovnou muzes soupnout defaultne NullLogger at tam nejsou ty podminky